ports/servoshell/egl/ohos/simpleservo.rs

```diff
@@ -23,6 +26,55 @@ use crate::egl::ohos::InitOpts;
 use crate::egl::ohos::resources::ResourceReaderInstance;
 use crate::prefs::{ArgumentParsingResult, parse_command_line_arguments};
 
+#[derive(Debug)]
+struct NativeValues {
+    cache_dir: String,
+    display_density: f32,
+    device_type: ohos_deviceinfo::OhosDeviceType,
+    os_full_name: String,
+}
+
+fn get_c_string_from_buffer_writer<E>(
+    f: unsafe extern "C" fn(*mut c_char, i32, *mut i32) -> Result<(), E>,
```

Comment:
This is probably not any `E`, but basically a `NonZeroI32` right? 
Does it even make sense to have this generic function, if we only have one user?